sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/GuidUtilities.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Azure.Messaging.ServiceBus.Primitives;
+
+internal static class GuidUtilities
+{
+    /// <summary>
+    /// The size, in bytes, to use for extracting the delivery tag bytes into <see cref="Guid"/>.
+    /// </summary>
+    private const int GuidSizeInBytes = 16;
+
+    public static Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
+    {
+        if (bytes.Length == GuidSizeInBytes)
+        {
+            // Use TryRead to avoid allocating an array if we are on a little endian machine.
+            if (!BitConverter.IsLittleEndian || !MemoryMarshal.TryRead<Guid>(bytes.Span, out var lockTokenGuid))
```

Comment:
It may be helpful to offer a bit more context around the approach and how endianness plays into it.   Would be especially helpful if this does move to Core.